src/Downloader.cc

```diff
@@ -127,6 +127,15 @@ downloaderDetach(clientStreamNode * node, ClientHttpRequest * http)
     clientStreamDetach(node, http);
 }
 
+static const char *
+debugPrintHeaders(const HttpHeader &h)
+{
+    static MemBuf tmp;
+    tmp.init();
+    h.packInto(&tmp, true);
+    return tmp.buf;
+}
```

Comment:
Please add an std::ostream `<<` operator to print HttpHeader instead. While implementing HttpHeader::print() or similar, please avoid legacy MemBuf and returning local/static buffers. Ideally, we should provide a zero-copy ostream-to-Packable converter (let me know if you are willing to wait for me to post a corresponding PR), but I will not insist on such an out-of-scope detour.

Reply:
I considered that approach at first. However, like you have realized HttpHeader really needs a different packInto() API and that is significant scope creep.

For now I made this static function to match all other header printing code. This Downloader is the oddity among HTTP message senders as all other callers generate the MemBuf which can be passed to debugs().

Reply:
Do you mind if I try to address my change request without introducing a lot of out-of-scope changes? If I succeed, this PR will not add dangerous code (that also uses legacy APIs that we want to get rid of). I think these benefits warrant such an attempt.